cmd/semv/main.go

```diff
@@ -0,0 +1,139 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"html/template"
+	"os"
+	"strings"
+
+	"github.com/rancher/ecm-distro-tools/release/semver"
+	"gopkg.in/yaml.v3"
+)
+
+func Format(v *semver.Version, format string) (string, error) {
+	switch {
+	case format == "":
+		str := fmt.Sprintf(
+			"Version: %s\nMajor: %d\nMinor: %d\nPatch: %d\nPrerelease: %s\nBuild: %s\n",
+			v.Version,
+			v.Major,
+			v.Minor,
+			v.Patch,
+			v.Prerelease,
+			v.Build,
+		)
+		return str, nil
+	case format == "json":
+		jsonData, err := json.MarshalIndent(v, "", "  ")
+		if err != nil {
+			return "", err
+		}
+		return string(jsonData), nil
+	case format == "yaml":
+		yml, err := yaml.Marshal(v)
+		if err != nil {
+			return "", err
+		}
+		return string(yml), nil
+	case strings.HasPrefix(format, "go-template="):
+		goTemplate := strings.TrimPrefix(format, "go-template=")
+		tmpl, err := template.New("output").Parse(goTemplate)
+		if err != nil {
+			return "", err
+		}
+		var buf bytes.Buffer
+		if err := tmpl.Execute(&buf, v); err != nil {
+			return "", err
+		}
+		return buf.String(), nil
+	}
+	return "", errors.New("invalid output format")
+}
+
+var (
+	name    string
+	version string
+	gitSHA  string
+)
+
+const usage = `version: %s
+Usage: %[2]s [-test] [-parse]
+Options:
+    -h            help
+    -v            show version and exit
+    -test         test a complete version against a semantic version pattern
```

Comment:
Can we have these flags as `-t` and below, `-p`?

Reply:
Refactored to use urfave with subcommands parse and test